doc/rst/technotes/nilChecking.rst

```diff
@@ -0,0 +1,34 @@
+.. _readme-nil-checking:
+
+=============================
+Checking for Nil Dereferences
+=============================
+
+The Chapel compiler includes two different mechanisms to check for nil
+dereferences. First, at compile-time, the compiler looks for places where
+it can prove a `nil` will be dereferenced or a method called on `nil`.
+When it can find such a case, it raises an error.
+
+The second mechanism is to check each at each method called on an object
+that the object is not `nil`. This works at run-time using checking code
```

Comment:
This sentence feels like it has too many "each"es.  (And maybe note earlier that this second mechanism is at run-time, like the first thing said about the first mechanism was that it's at compile-time?)